From 67d8ba8f8528b9c217cd0177c5e362eba9155d4a Mon Sep 17 00:00:00 2001
From: Amelie Delaunay <amelie.delaunay@foss.st.com>
Date: Tue, 28 Jan 2025 13:09:53 +0100
Subject: [PATCH 1507/1695] dts: stm32: fix I2C3 pinmux for stm32mp215f-dk

Pinmux for I2C3 belongs to pinctrl_z node, not pinctrl. Move i2c3_pins_a
under pinctrl_z node.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git c1d0d3743e7527b69b48c34799bac3fa1af68efe dts: stm32: fix I2C3 pinmux for stm32mp215f-dk]
Fixes: fef4652da1f7 ("dts: stm32: add I2C3 pinmux for stm32mp215f-dk")
Signed-off-by: Amelie Delaunay <amelie.delaunay@foss.st.com>
Change-Id: I19397696213cc9635fef65f6cae5318bfe2381b2
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/432239
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/arch/arm/dts/stm32mp21-pinctrl.dtsi | 22 ++++++++++++----------
 1 file changed, 12 insertions(+), 10 deletions(-)

diff --git a/core/arch/arm/dts/stm32mp21-pinctrl.dtsi b/core/arch/arm/dts/stm32mp21-pinctrl.dtsi
index 1a7773ebe..1186bdcf2 100644
--- a/core/arch/arm/dts/stm32mp21-pinctrl.dtsi
+++ b/core/arch/arm/dts/stm32mp21-pinctrl.dtsi
@@ -7,16 +7,6 @@
 #include <dt-bindings/pinctrl/stm32-pinfunc.h>
 
 &pinctrl {
-	i2c3_pins_a: i2c3-0 {
-		pins1 {
-			pinmux = <STM32_PINMUX('Z', 1, AF8)>, /* I2C3_SCL */
-				 <STM32_PINMUX('Z', 0, AF8)>; /* I2C3_SDA */
-			bias-disable;
-			drive-open-drain;
-			slew-rate = <0>;
-		};
-	};
-
 	ltdc_pins_a: ltdc-0 {
 		pins {
 			pinmux = <STM32_PINMUX('F', 12, AF13)>, /* LCD_CLK */
@@ -60,3 +50,15 @@
 		};
 	};
 };
+
+&pinctrl_z {
+	i2c3_pins_a: i2c3-0 {
+		pins1 {
+			pinmux = <STM32_PINMUX('Z', 1, AF8)>, /* I2C3_SCL */
+				 <STM32_PINMUX('Z', 0, AF8)>; /* I2C3_SDA */
+			bias-disable;
+			drive-open-drain;
+			slew-rate = <0>;
+		};
+	};
+};
-- 
2.47.2

